app/controllers/tasks_controller.rb

```diff
@@ -5,7 +5,13 @@ class TasksController < ApplicationController
   before_action :check_user, only: %i[edit update]
 
   def index
-    @tasks = Task.order(:created_at).page(params[:page])
+    if params[:search]
```

Comment:
Let's try this refactoring instead of calculation on temp values.

```diff
diff --git a/app/controllers/tasks_controller.rb b/app/controllers/tasks_controller.rb
index 5e96194..9b6a852 100644
--- a/app/controllers/tasks_controller.rb
+++ b/app/controllers/tasks_controller.rb
@@ -5,13 +5,13 @@ class TasksController < ApplicationController
   before_action :check_user, only: %i[edit update]
 
   def index
-    if params[:search]
-      @tasks_by_status = status_params.nil? ? Task.all : Task.search_by_status(status_params)
-      @tasks_by_status_and_user = user_id_params.nil? ? @tasks_by_status : @tasks_by_status.search_by_user_id(user_id_params)
-      @tasks = @tasks_by_status_and_user.order(:created_at).page(params[:page])
-    else
-      @tasks = Task.order(:created_at).page(params[:page])
-    end
+    @tasks =
+      current_user
+      .tasks
+      .then(&method(:filter_by_status))
+      .then(&method(:filter_by_user_id))
+      .then(&method(:order))
+      .then(&method(:paginate))
   end
 
   def show
@@ -69,4 +69,28 @@ class TasksController < ApplicationController
   def user_id_params
     params.require(:search).permit(:user_id)[:user_id]
   end
+
+  def filter_by_status(tasks)
+    if status_params.present?
+      tasks.search_by_status(status_params)
+    else
+      tasks
+    end
+  end
+
+  def filter_by_user_id(tasks)
+    if user_id_params.present?
+      tasks.search_by_user_id(user_id_params)
+    else
+      tasks
+    end
+  end
+
+  def order(tasks)
+    tasks.order(:created_at)
+  end
+
+  def paginate(tasks)
+    tasks.page(params[:page])
+  end
 end
```

Reply:
Ref about how we can compose chainable methods. https://thoughtbot.com/blog/using-yieldself-for-composable-activerecord-relations